src/main/java/org/isf/stat/reportlauncher/gui/ReportLauncher.java

```diff
@@ -234,9 +219,53 @@ private JPanel getJParameterSelectionPanel() {
 			JLabel jRptLabel = new JLabel(MessageBundle.getMessage("angal.stat.report"));
 
 			jRptComboBox = new JComboBox<>();
-			for (String[] matrix : reportMatrix) {
-				jRptComboBox.addItem(MessageBundle.getMessage(matrix[BUNDLE]));
-			}
+			jasperFilesInFolder = new LinkedList<>();
+			folderNameFileNameMap = new HashMap<>();
+			try {				
+				List<File> loadedFiles = Files.walk(Paths.get("./rpt_stat"))
+								.filter(Files::isRegularFile)
+                                .map(Path::toFile)
+                                .filter(t -> t.getName().endsWith(".jasper"))
+                                .collect(Collectors.toList());
+				jasperFilesInFolder.addAll(loadedFiles);
+				folderNameFileNameMap.put("rpt_stat", loadedFiles.stream().map(t -> t.getName().replace(".jasper", "")).collect(Collectors.toList()));
+			
+				loadedFiles = Files.walk(Paths.get("./rpt_extra"))
+                                .filter(Files::isRegularFile)
+                                .map(Path::toFile)
+                                .filter(t -> t.getName().endsWith(".jasper"))
+                                .collect(Collectors.toList());
+				jasperFilesInFolder.addAll(loadedFiles);
+				folderNameFileNameMap.put("rpt_extra", loadedFiles.stream().map(t -> t.getName().replace(".jasper", "")).collect(Collectors.toList()));
+			
+				reportNameFileMap = new HashMap<>();
+				for(File f: jasperFilesInFolder) {
+					try {
+						Properties props = ConfigurationProperties.loadPropertiesFile(
+										f.getName().replace(".jasper", "_"+(new Locale(GeneralData.LANGUAGE)).getLanguage()+".properties"), LOGGER);
+						
+						if(props != null && props.getProperty("jTitle") != null && !props.getProperty("jTitle").isEmpty()) {
+							reportNameFileMap.put(props.getProperty("jTitle"), f);
+							jRptComboBox.addItem(props.getProperty("jTitle"));
+						}
+						else {
+							props = ConfigurationProperties.loadPropertiesFile(
+											f.getName().replace(".jasper", ".properties"), LOGGER);
+							
+							if(props != null && props.getProperty("jTitle") != null && !props.getProperty("jTitle").isEmpty()) {
+								reportNameFileMap.put(props.getProperty("jTitle"), f);
+								jRptComboBox.addItem(props.getProperty("jTitle"));
+							}
```

Comment:
looks like if jTitle is missing in all .properties, the report is not shown nor added to the list (differently as stated in the doc). Nevertheless, this could be good in order to force the presence of jTitle key.

Anyway, what we should be able to achieve is to show them also in alphabetical order

![image](https://user-images.githubusercontent.com/2938553/213914695-f552f2e3-6004-46cc-8e50-75f0394694d1.png)

Reply:
If jTitle is missing in both localized and default .properties, now the report is added to the list with the name of the report itself and the report list is now shown in alphabetical order